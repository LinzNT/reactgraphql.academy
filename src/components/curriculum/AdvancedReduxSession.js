import React from 'react'
import Ul, { Li } from '../layout/Ul'
import Session from './Session'

const AdvancedReduxSession = ({ title }) => (
  <Session title={title}>
    <Ul>
      <Li>
        Functional programming (FP)
        <Ul>
          <Li>Composing functions</Li>
          <Li>Currying, thunks and higher-order functions</Li>
        </Ul>
      </Li>
      <Li>
        Advanced Redux:
        <Ul>
          <Li>Introducing the Redux middleware</Li>
          <Li>Exercise, build your own Redux middleware</Li>
          <Li>Thunks. Actions that dispatch actions</Li>
          <Li>Exercise, implement loading indicators using thunks</Li>
        </Ul>
      </Li>
      <Li>Time travel in Redux using Redux DevTools</Li>
    </Ul>
  </Session>
)

export default AdvancedReduxSession
