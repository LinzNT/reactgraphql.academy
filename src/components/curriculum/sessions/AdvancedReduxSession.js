import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

const AdvancedReduxSession = ({ title }) => (
  <Session title={title}>
    <Ul>
      <Li>
        Functional programming (FP)
        <Ul>
          <Li>Composing functions</Li>
          <Li>Currying, thunks and higher-order functions</Li>
        </Ul>
      </Li>
      <Li>
        Advanced Redux:
        <Ul>
          <Li>
            Using the Redux middleware, and building your own Redux middlewares
          </Li>
          <Li>
            Thunks, actions that dispatch actions. Implementing loading
            indicators using thunks
          </Li>
          <Li>
            Using selectors to write more robust and performant code. Learn how{' '}
            <code>reselect</code> works
          </Li>
          <Li>
            Sagas Vs thunks. Exercise, reimplement the previous thunk loading
            indicator with <code>redux-saga</code>.
          </Li>
        </Ul>
      </Li>
      <Li>Time travel in Redux using Redux DevTools</Li>
    </Ul>
  </Session>
)

export default AdvancedReduxSession
