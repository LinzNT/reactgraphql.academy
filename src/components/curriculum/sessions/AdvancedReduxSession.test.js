import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import AdvancedReduxSession from './AdvancedReduxSession'

describe('<AdvancedReduxSession />', () => {
  it('should render component', () => {
    const wrapper = shallow(<AdvancedReduxSession />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
