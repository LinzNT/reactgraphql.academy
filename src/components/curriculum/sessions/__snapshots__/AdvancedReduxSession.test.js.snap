// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedReduxSession /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      Functional programming (FP)
      <styled.ul>
        <styled.li>
          Composing functions
        </styled.li>
        <styled.li>
          Currying, thunks and higher-order functions
        </styled.li>
      </styled.ul>
    </styled.li>
    <styled.li>
      Advanced Redux:
      <styled.ul>
        <styled.li>
          Using the Redux middleware, and building your own Redux middlewares
        </styled.li>
        <styled.li>
          Thunks, actions that dispatch actions. Implementing loading indicators using thunks
        </styled.li>
        <styled.li>
          Using selectors to write more robust and performant code. Learn how
           
          <code>
            reselect
          </code>
           works
        </styled.li>
        <styled.li>
          Sagas Vs thunks. Exercise, reimplement the previous thunk loading indicator with 
          <code>
            redux-saga
          </code>
          .
        </styled.li>
      </styled.ul>
    </styled.li>
    <styled.li>
      Time travel in Redux using Redux DevTools
    </styled.li>
  </styled.ul>
</Session>
`;
